src/bitmap/bitmap_8k.rs

```diff
@@ -0,0 +1,314 @@
+
+    pub fn is_disjoint(&self, other: &Bitmap8K) -> bool {
+        self.bits.iter().zip(other.bits.iter()).all(|(&i1, &i2)| (i1 & i2) == 0)
+    }
+
+    pub fn is_subset(&self, other: &Self) -> bool {
+        self.bits.iter().zip(other.bits.iter()).all(|(&i1, &i2)| (i1 & i2) == i1)
+    }
+
+    pub fn to_array_store(&self) -> Store {
+        let mut vec = Vec::with_capacity(self.len as usize);
+        for (index, mut bit) in self.bits.iter().cloned().enumerate() {
+            while bit != 0 {
+                vec.push((u64::trailing_zeros(bit) + (64 * index as u32)) as u16);
+                bit &= bit - 1;
+            }
+        }
+        Store::Array(SortedU16Vec::from_vec(vec))
+    }
+
+    pub fn len(&self) -> u64 {
+        self.len
+    }
+
+    pub fn min(&self) -> Option<u16> {
+        self.bits
+            .iter()
+            .enumerate()
+            .find(|&(_, &bit)| bit != 0)
+            .map(|(index, bit)| (index * 64 + (bit.trailing_zeros() as usize)) as u16)
+    }
+
+    pub fn max(&self) -> Option<u16> {
+        self.bits
+            .iter()
+            .enumerate()
+            .rev()
+            .find(|&(_, &bit)| bit != 0)
+            .map(|(index, bit)| (index * 64 + (63 - bit.leading_zeros() as usize)) as u16)
+    }
+
+    pub fn iter(&self) -> BitmapIter<&[u64; BITMAP_LENGTH]> {
+        BitmapIter::new(&self.bits)
+    }
+
+    pub fn into_iter(self) -> BitmapIter<Box<[u64; BITMAP_LENGTH]>> {
+        BitmapIter::new(self.bits)
+    }
+
+    pub fn as_array(&self) -> &[u64; BITMAP_LENGTH] {
+        &self.bits
+    }
+}
+
+pub struct BitmapIter<B: Borrow<[u64; BITMAP_LENGTH]>> {
+    key: usize,
+    bit: usize,
+    bits: B,
+}
+
+impl<B: Borrow<[u64; BITMAP_LENGTH]>> BitmapIter<B> {
+    fn new(bits: B) -> BitmapIter<B> {
+        BitmapIter { key: 0, bit: 0, bits }
+    }
+
+    fn move_next(&mut self) {
+        self.bit += 1;
+        if self.bit == 64 {
+            self.bit = 0;
+            self.key += 1;
+        }
+    }
+}
+
+impl<B: Borrow<[u64; BITMAP_LENGTH]>> Iterator for BitmapIter<B> {
+    type Item = u16;
+
+    fn next(&mut self) -> Option<u16> {
+        loop {
+            if self.key == BITMAP_LENGTH {
+                return None;
+            } else if (unsafe { self.bits.borrow().get_unchecked(self.key) } & (1u64 << self.bit))
+                != 0
+            {
+                let result = Some((self.key * 64 + self.bit) as u16);
+                self.move_next();
+                return result;
+            } else {
+                self.move_next();
+            }
+        }
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        panic!("Should never be called (roaring::Iter caches the size_hint itself)")
+    }
+}
+
+#[inline]
+pub fn key(index: u16) -> usize {
+    index as usize / 64
+}
+
+#[inline]
+pub fn bit(index: u16) -> usize {
+    index as usize % 64
+}
+
+#[inline]
+fn op_bitmaps(bits1: &mut Bitmap8K, bits2: &Bitmap8K, op: fn(&mut u64, u64)) {
```

Comment:
Why using an `fn()` here and not letting Rust optimize this maybe better by using an `Fn` trait?
```suggestion
fn op_bitmaps<F: Fn(&mut u64, u64)>(bits1: &mut Bitmap8K, bits2: &Bitmap8K, op: F) {
```

Also, don't you think that using a function that does not modify its parameters would be better optimized? Something like `Fn(u64, u64) -> u64`?

Reply:
Today I learned `fn` is a function pointer and `Fn` is a monomophizable trait. Yes! Great catch!